FpsCheck.py

```diff
@@ -103,17 +103,14 @@ def has_all_rationales(self, check_sets):
         error_list
 
         Args:
-            check_sets: a dictionary of primary->FpsSet
+            None
         Returns:
             None
         """
         for fpset in self.fps_sites['sets']:
-            sites = check_sets[fpset.get("primary")].associated_sites
-            service_sites = check_sets[fpset.get("primary")].service_sites
-            if sites:
-                sites = sites + service_sites if service_sites else sites
-            else:
-                sites = service_sites
+            if fpset.get("primary") not in  check_sets:
+                continue
```

Comment:
Should we just loop over `check_sets` instead? It's not clear to me how/if the data in `self.fps_sites['sets']` is different from `check_sets`, other than being a superset of that dict.

Reply:
Since the rationales aren't stored in FpsSet objects, we need to get the rationales from the JSON document stored at `self.fps_sites` when we run `has_all_rationales`. I thought it would be better to iterate through the list of sets stored at `self.fps_sites['sets']` and check for their presence in `check_sets` rather than vice versa because `check_sets` is a dictionary while `self.fps_sites['sets']` is a list. So the current implementation should have time complexity O(N) while doing the alternative would be O(N<sup>2</sup>).

Best practice is probably to re-write the FpsSet class to incorporate the rationales in a variable, but I would like to do that in a future PR when the time comes.

Reply:
Agreed, I think we ought to make the script rely entirely on the `FpsSet` instances, rather than a mixture of the `FpsSet`s and the raw dictionaries. That can be done in a followup, though.